cpp/src/arrow/compare.cc

```diff
@@ -1136,19 +1137,60 @@ namespace {
 template <typename LeftSparseIndexType, typename RightSparseIndexType>
 struct SparseTensorEqualsImpl {
   static bool Compare(const SparseTensorImpl<LeftSparseIndexType>& left,
-                      const SparseTensorImpl<RightSparseIndexType>& right) {
+                      const SparseTensorImpl<RightSparseIndexType>& right,
+                      const EqualOptions&) {
     // TODO(mrkn): should we support the equality among different formats?
     return false;
   }
 };
 
+bool IntegerSparseTensorDataEquals(const uint8_t* left_data, const uint8_t* right_data,
+                                   const int byte_width, const int64_t length) {
+  if (left_data == right_data) {
+    return true;
+  }
+  return memcmp(left_data, right_data, static_cast<size_t>(byte_width * length)) == 0;
+}
+
+template <typename DataType>
+bool FloatSparseTensorDataEquals(const typename DataType::c_type* left_data,
```

Comment:
I renamed `Floating` to `Float` here and other places in this pull-request.
But I didn't rename `Floating` outside of this pull-request, such as `BaseFloatingEquals` at [the top of this file](https://github.com/apache/arrow/blob/f18acac8be618a59378ce01694983430042f870b/cpp/src/arrow/compare.cc#L62).